app/harvesters/abstract_harvester.py

```diff
@@ -105,16 +105,18 @@ async def run(self) -> None:
         )
         existing_references: list[Reference] = []
         try:
-            result: AbstractHarvesterRawResult
-            async for result in self.fetch_results():
-                if result is None or result == "end":
+            raw_data: AbstractHarvesterRawResult
+            async for raw_data in self.fetch_results():
+                if raw_data is None or raw_data == "end":
                     break
-                new_ref = await self.converter.convert(result)
+                new_ref = self.converter.build(raw_data)
                 if new_ref is None:
                     continue
                 old_ref = await references_recorder.exists(new_ref=new_ref)
                 if old_ref is not None:
                     existing_references.append(old_ref)
+                if (old_ref is None) or (new_ref.hash != old_ref.hash):
+                    await self.converter.convert(raw_data=raw_data, new_ref=new_ref)
```

Comment:
@Paldana99 @E-Bara From now we don't convert unless the reference is new or has changed.
"convert()" method returns nothing : the reference is built by a "build()" method, which has been factorized at the AbstractConverter level. "build()" only populates source identifier, harvester and hash.